src/ApplicationInsights.Kubernetes/K8sQueryClient.cs

```diff
@@ -51,16 +56,30 @@ public Task<IEnumerable<K8sPod>> GetPodsAsync()
         {
             EnsureNotDisposed();
 
-            string myContainerId = KubeHttpClient.Settings.ContainerId;
-            IEnumerable<K8sPod> possiblePods = await GetPodsAsync().ConfigureAwait(false);
-            if (possiblePods == null)
+            IEnumerable<K8sPod> allPods = await GetPodsAsync().ConfigureAwait(false);
+            if (allPods == null)
             {
                 return null;
             }
-            K8sPod targetPod = possiblePods.FirstOrDefault(pod => pod.Status != null &&
-                                pod.Status.ContainerStatuses != null &&
-                                pod.Status.ContainerStatuses.Any(
-                                    cs => !string.IsNullOrEmpty(cs.ContainerID) && cs.ContainerID.EndsWith(myContainerId, StringComparison.Ordinal)));
+
+            K8sPod targetPod = null;
+            string podName = Environment.GetEnvironmentVariable(@"APPINSIGHTS_KUBERNETES_POD_NAME");
+            string myContainerId = KubeHttpClient.Settings.ContainerId;
+            if (!string.IsNullOrEmpty(podName))
+            {
+                targetPod = allPods.FirstOrDefault(p => p.Metadata.Name.Equals(podName, StringComparison.Ordinal));
```

Comment:
Opportunity for improvement: if you know the pod name, use it in the query. In a large system there might be a lot of pods!

Reply:
Good point. Filed a new issue: https://github.com/Microsoft/ApplicationInsights-Kubernetes/issues/114